### PR TITLE
[MOB-975] Complete google pay one off payment

### PR DESCRIPTION
**Pull request by aw-hector**
Complete one-off payment flow by:

- Get Google Pay payment sheet displayed by calling loadPaymentData request
- Transform and forward the google pay success response to PA backend to confirm the payment intent

Some chores:
- Make merchant ID configurable in demo app
- Fix issue where CI not correctly loads custom detekt config
- Connects staging & demo to google pay Test environment while prod connects to Prod

### CODE REVIEW

File: components-core/src/main/java/com/airwallex/android/core/Airwallex.kt
Comment (jarrodrobins) [inline components-core/src/main/java/com/airwallex/android/core/Airwallex.kt:363]: This method is pretty huge. Can we split this out into multiple methods? 

Also this is probably a stylistic choice but I feel like being explicit with if/else is easier to read than using `let` with `?:` as an else case.

File: googlepay/src/main/java/com/airwallex/android/googlepay/GooglePayComponent.kt
Comment (jarrodrobins) [inline googlepay/src/main/java/com/airwallex/android/googlepay/GooglePayComponent.kt:91]: Yeah these `lets` are very hard to read :P. Can we improve this?

File: components-core/src/main/java/com/airwallex/android/core/Airwallex.kt
Comment (aw-hector) [reply components-core/src/main/java/com/airwallex/android/core/Airwallex.kt:363]: Good idea! extracted to the separate method `checkoutGooglePay`

File: googlepay/src/main/java/com/airwallex/android/googlepay/GooglePayComponent.kt
Comment (aw-hector) [reply googlepay/src/main/java/com/airwallex/android/googlepay/GooglePayComponent.kt]: yep, just refactored using `if else`

**Comment by aw-hector** [comment]
Sonarqube fails at coverage after we update the main branch to develop even this PR's code changes have got above 80% coverage, may need to force merge it for this after getting approval.

**Review by JinW-Airwallex — APPROVED** [review]

**Comment by JinW-Airwallex** [comment]
Nice work Hector!

**Pull request merged**
